packages/react/src/useSignTransaction.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    SolanaSignAndSendTransaction,
+    SolanaSignTransaction,
+    SolanaSignTransactionFeature,
+    SolanaSignTransactionInput,
+    SolanaSignTransactionOutput,
+} from '@solana/wallet-standard-features';
+import {
+    WALLET_STANDARD_ERROR__FEATURES__WALLET_ACCOUNT_CHAIN_UNSUPPORTED,
+    WalletStandardError,
+} from '@wallet-standard/errors';
+import { getWalletAccountFeature, UiWalletAccount } from '@wallet-standard/ui';
+import { getWalletAccountForUiWalletAccount_DO_NOT_USE_OR_YOU_WILL_BE_FIRED } from '@wallet-standard/ui-registry';
+import { useCallback } from 'react';
+
+import { OnlySolanaChains } from './chain';
+
+type Input = Omit<SolanaSignTransactionInput, 'account' | 'chain'>;
+type Output = SolanaSignTransactionOutput;
+
+/**
+ * Returns a function you can call to sign a serialized transaction.
+ */
+export function useSignTransaction<TWalletAccount extends UiWalletAccount>(
+    ...config: Parameters<typeof useSignTransactions<TWalletAccount>>
+): (input: Input) => Promise<Output> {
+    const signTransactions = useSignTransactions(...config);
+    return useCallback(
+        async input => {
+            const [result] = await signTransactions(input);
+            return result;
+        },
+        [signTransactions],
+    );
+}
+
+/**
+ * The plural version of `useSignTransaction`. Call this function with more than one input to sign
+ * multiple transactions in one operation.
+ *
+ * See `useSignTransaction()` for usage.
+ */
+export function useSignTransactions<TWalletAccount extends UiWalletAccount>(
+    uiWalletAccount: TWalletAccount,
+    chain: OnlySolanaChains<TWalletAccount['chains']>,
+): (...inputs: readonly Input[]) => Promise<readonly Output[]> {
+    if (!uiWalletAccount.chains.includes(chain)) {
+        throw new WalletStandardError(WALLET_STANDARD_ERROR__FEATURES__WALLET_ACCOUNT_CHAIN_UNSUPPORTED, {
+            address: uiWalletAccount.address,
+            chain,
+            featureName: SolanaSignAndSendTransaction,
+            supportedChains: [...uiWalletAccount.chains],
+            supportedFeatures: [...uiWalletAccount.features],
+        });
+    }
+    const signTransactionFeature = getWalletAccountFeature(
+        uiWalletAccount,
+        SolanaSignTransaction,
+    ) as SolanaSignTransactionFeature[typeof SolanaSignTransaction];
```

Comment:
Just for my understanding: The docs say that these inputs dictate how a wallet should simulate a transaction (min context slot, etc.). Is this where wallets plug in that simulation?

Reply:
I've stripped this out of the docs and the inputs. Wallets decide; apps have no say.